chromadb/utils/batch_utils.py

```diff
@@ -1,36 +1,64 @@
-from typing import Optional, Tuple, List
-from chromadb.api import BaseAPI
-from chromadb.api.types import (
-    Documents,
-    Embeddings,
-    IDs,
-    Metadatas,
-)
+from typing import Any, Iterator, List, Optional, Tuple, TypeVar, Union, cast
+from chromadb.api import ClientAPI
+import numpy as np
+from rich.progress import track
+
+# (Can't provide type arguments to ndarray < 3.9.)
+T = TypeVar("T", bound=Tuple[Union[List[Any], np.ndarray, None], ...])  # type: ignore[type-arg]
 
 
 def create_batches(
-    api: BaseAPI,
-    ids: IDs,
-    embeddings: Optional[Embeddings] = None,
-    metadatas: Optional[Metadatas] = None,
-    documents: Optional[Documents] = None,
-) -> List[Tuple[IDs, Embeddings, Optional[Metadatas], Optional[Documents]]]:
-    _batches: List[
-        Tuple[IDs, Embeddings, Optional[Metadatas], Optional[Documents]]
-    ] = []
-    if len(ids) > api.get_max_batch_size():
-        # create split batches
-        for i in range(0, len(ids), api.get_max_batch_size()):
-            _batches.append(
-                (  # type: ignore
-                    ids[i : i + api.get_max_batch_size()],
-                    embeddings[i : i + api.get_max_batch_size()]
-                    if embeddings
-                    else None,
-                    metadatas[i : i + api.get_max_batch_size()] if metadatas else None,
-                    documents[i : i + api.get_max_batch_size()] if documents else None,
-                )
-            )
-    else:
-        _batches.append((ids, embeddings, metadatas, documents))  # type: ignore
-    return _batches
+    client: ClientAPI,
+    records: T,
+    max_batch_size: Optional[int] = 1024,
+    print_progress_description: Optional[str] = None,
```

Comment:
for progress could we tqdm if in a notebook?

Reply:
sorry, not sure what you mean, this implementation includes a progress bar using the rich module
rich [works in notebooks](https://rich.readthedocs.io/en/stable/progress.html)